soc/arm/nxp_imx/rt/soc_rt10xx.c

```diff
@@ -292,6 +292,11 @@ static int imxrt_init(void)
 #ifndef CONFIG_IMXRT1XXX_CODE_CACHE
 	/* SystemInit enables code cache, disable it here */
 	SCB_DisableICache();
+#else
+	/* z_arm_init_arch_hw_at_boot() disables code cache if CONFIG_ARCH_CACHE is enabled,
+	 * enable it here.
```

Comment:
My understanding from this comment is that we have to enable code cache _only_ if CONFIG_ARCH_CACHE is set.
Is this always set, or why don't we check for this config before doing SCB_EnableICache?

Also, should be done the same thing here: https://github.com/zephyrproject-rtos/zephyr/blob/main/soc/arm/nxp_imx/rt/soc_rt11xx.c#L659 ?

Reply:
> My understanding from this comment is that we have to enable code cache _only_ if CONFIG_ARCH_CACHE is set. Is this always set, or why don't we check for this config before doing SCB_EnableICache?

That is currently true. However, if anyone were to change `z_arm_init_arch_hw_at_boot` without considering what's happening here (that code affects all ARM SOCs, not only RT10xx), we might end up with a disabled instruction cache again.

As @danieldegrasse mentioned above, there is no harm in calling `SCB_EnableICache` when the instruction cache is enabled already, as the function first checks for that and just returns. On the other hand, if the instruction cache is not enabled at this point for any reason whatsoever, I want it to be enabled ASAP or my application is slowed down to a crawl again.

> Also, should be done the same thing here: https://github.com/zephyrproject-rtos/zephyr/blob/main/soc/arm/nxp_imx/rt/soc_rt11xx.c#L659 ?

Unfortunately I don't have any RT11xx hardware to test this, but it looks to me like you are correct. I'm just not sure if I should add code that I can't test.

Reply:
> Also, should be done the same thing here: [main/soc/arm/nxp_imx/rt/soc_rt11xx.c#L659](https://github.com/zephyrproject-rtos/zephyr/blob/main/soc/arm/nxp_imx/rt/soc_rt11xx.c?rgh-link-date=2023-05-03T10%3A17%3A36Z#L659) ?

Yes, likely so. I'll open a PR with this fix, as I have an RT11xx available to verify it with